src/place-screen/hooks/use-quay-data.ts

```diff
@@ -293,6 +293,16 @@ export function useQuayData(
     [],
     !isFocused || mode !== 'Departure',
   );
+  useEffect(() => {
+    if (!isFocused || !state.tick) return;
+
+    const timeSinceLastTick = differenceInSeconds(Date.now(), state.tick);
+    if (timeSinceLastTick / 60 >= HARD_REFRESH_LIMIT_IN_MINUTES) {
+      loadDepartures();
+    } else if (timeSinceLastTick >= updateFrequencyInSeconds) {
+      dispatch({type: 'LOAD_REALTIME_DATA', quay});
+    }
+  }, [isFocused]);
```

Comment:
These looks exactly the same with the same goal? I think if so this is the perfect opportunity to create a new custom hook? I think in general we can be better at having smaller custom hooks

Reply:
I though about it, but also thought it would be messy with all the params. But tried it now, and it turned out a lot cleaner ✨